samples/cloud-client/snippets/create_api_key.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START auth_cloud_create_api_key]
+
+from google.cloud import api_keys_v2
+from google.cloud.api_keys_v2 import Key
+
+
+def create_api_key(project_id: str, location: str) -> Key:
+    """
+    Create and restrict an API key.
+
+    // TODO(Developer):
+    //  1. Before running this sample,
+    //  set up ADC as described in https://cloud.google.com/docs/authentication/external/set-up-adc
+    //  2. Make sure you have the necessary permission to create API keys.
+
+    Args:
+        project_id: Google Cloud project id.
+        location: Can only be "global".
+    """
+
+    # Create the API Keys client.
+    client = api_keys_v2.ApiKeysClient()
+
+    # Restrict the API key usage by specifying the target service and methods.
+    # The API key can only be used to authenticate the specified methods in the service.
+    api_target = api_keys_v2.ApiTarget()
+    api_target.service = "translate.googleapis.com"
+    api_target.methods = ["transate.googleapis.com.TranslateText"]
+
+    # Set the API restriction.
```

Comment:
Would it be possible to just create the key here, and then create snippets for the different types of restrictions? As it is organized in the doc page? https://cloud.google.com/docs/authentication/api-keys#adding-application-restrictions
Would be good to put in a note saying "you should always restrict your keys" here, but it would help me include the snippets if they followed the organization of the page more closely.

Reply:
Pia, I've modified the samples to account for the doc page structure. PTAL.